server/controllers/request.js

```diff
@@ -0,0 +1,90 @@
+const Request = require("../Models/Request");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /request
+// @desc List of requests for logged in user
+// @access Private
+exports.getRequests = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+
+  const requests = await Request.find({
+    $or: [{ userId: userId }, { sitterId: userId }],
+  });
+
+  res.send({ requests });
+});
+
+// @route POST /request
+// @desc Create a new request
+// @access Private
+exports.createRequest = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+
+  const { sitterId, startDate, endDate } = req.body;
+
+  const sitterExists = await User.findById(sitterId);
+
+  if (!sitterExists) {
+    res.status(400);
+    throw new Error("Invalid sitter id");
+  }
+
+  if (!startDate || !endDate) {
+    res.status(400);
+    throw new Error("Incomplete date field");
+  }
```

Comment:
I think this one should come before `sitterExists`, so that no need to spend time querying the database if invalid fields are suplied.